warp10/src/main/java/io/warp10/script/functions/SNAPSHOT.java

```diff
@@ -411,4 +421,47 @@ public static void addElement(SNAPSHOT snapshot, StringBuilder sb, Object o, boo
       }
     }
   }
+
+  //
+  // Process a string to make it readable and compatible in WarpScript code
+  //
+
+  private static String processString(String s) {
+
+    char[] chars = UnsafeString.getChars(s);
+
+    StringBuilder sb = null;
+
+    int lastIdx = 0;
+    int idx = 0;
+
+    //
+    // Replace anything below 32 and ' by %## (invalid character)
+    //
+
+    while(idx < chars.length) {
+      if ("'".charAt(0) == chars[idx] || chars[idx] < 32) {
+
+        if (null == sb) {
+          sb = new StringBuilder(s.length() + 2);
+        }
+
+        sb.append(chars, lastIdx, idx - lastIdx);
+        sb.append("%##");
```

Comment:
You need to output %hh where hh is the hexadecimal code of the char less than 32 you encountered.